internal/api/v1beta1/servicedata.go

```diff
@@ -132,7 +138,7 @@ func (h Handler) UpsertUserServiceData(ctx context.Context, request *shieldv1bet
 	}
 
 	return &shieldv1beta1.UpsertUserServiceDataResponse{
-		Data: serviceDataMap,
+		Data: requestBody.GetData(),
```

Comment:
This should be the servicedatamap pb from serviceDataMap right?
```suggestion
		Data: serviceDataMapPB,
```

where 
```
serviceDataMapPB, err := structpb.NewStruct(serviceDataMap)
if err != nil {
    return nil, grpcInternalServerError
}
```